src/scenes/Event/components/EventTxHistory/txRow.js

```diff
@@ -0,0 +1,92 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from 'react-intl';
+import {
+  TableCell,
+  TableRow,
+  withStyles,
+} from '@material-ui/core';
+import cx from 'classnames';
+
+import styles from './styles';
+import TransactionHistoryID from '../../../../components/TransactionHistoryAddressAndID/id';
+import TransactionHistoryAddress from '../../../../components/TransactionHistoryAddressAndID/address';
+import { getShortLocalDateTimeString } from '../../../../helpers/utility';
+import { localizeInvalidOption } from '../../../../helpers/localizeInvalidOption';
+import { getTxTypeString } from '../../../../helpers/stringUtil';
+import { TransactionType } from '../../../../constants';
+
+@injectIntl
+@withStyles(styles, { withTheme: true })
+export default class TxRow extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired, // eslint-disable-line react/no-typos
+    classes: PropTypes.object.isRequired,
+    transaction: PropTypes.object.isRequired,
+  };
+
+  state = {
+    expanded: false,
+  }
+
+  toggle = () => {
+    // probably need some logic here for not expanding when clicking on
+    // the Name link field
+    this.setState({ expanded: !this.state.expanded });
+  };
+
+  getDescription = (tx) => {
```

Comment:
let's make this a js `getter` like this
```js
get name() {
  const { optionIdx, topic, type, intl } = this.props
  // then we don't have to us "tx.whatever" everywhere
}
```
then we can use it below like
```js
<TableCell padding="dense">{this.name}</TableCell>
// instead of
<TableCell padding="dense">{this.getDescription(transaction)}</TableCell>
```